Sources/Imperial/Helpers/Sessions+Imperial.swift

```diff
@@ -38,7 +49,25 @@ extension Session {
     public func setAccessToken(_ token: String) {
         self[Keys.token] = token
     }
-    
+
+    /// Gets the refresh token from the session.
+    ///
+    /// - Returns: The refresh token stored with the `refresh_token` key.
+    /// - Throws: `Abort.unauthorized` if no refresh token exists.
+    public func refreshToken()throws -> String {
+        guard let token = self[Keys.refresh] else {
+            throw Abort(.unauthorized, reason: "User currently not authenticated")
```

Comment:
Depending on the OAuth provider, couldn't this value not exist and the user would still be authenticated?

Reply:
True! I pondered the same question today and will reword the error message.